### PR TITLE
Add type for Moran

### PR DESCRIPTION
**Pull request by souravsingh**
Fixes a part of #808

### CODE REVIEW

File: axelrod/moran.py [hunk 1]
```diff
@@ -107,7 +108,7 @@ def set_players(self):
             self.players.append(player)
         self.populations = [self.population_distribution()]
 
-    def mutate(self, index):
+    def mutate(self, index: int):
         """Mutate the player at index."""
         # Choose another strategy at random from the initial population
         r = random.random()
```
Comment (janga1997) [inline axelrod/moran.py]: Do add a return type hint to function.

File: axelrod/moran.py [hunk 2]
```diff
@@ -131,7 +132,7 @@ def death(self, index=None):
         i = randrange(0, len(self.players))
         return i
```
Comment (janga1997) [inline axelrod/moran.py]: Same here.

File: axelrod/moran.py [hunk 3]
```diff
@@ -334,7 +335,7 @@ def birth(self, index=None):
             j = fitness_proportionate_selection(scores)
         return j
```
Comment (janga1997) [inline axelrod/moran.py]: And here too.

File: axelrod/moran.py (hunk 1 above)
Comment (drvinceknight) [inline axelrod/moran.py]: > Do add a return type hint to function.

:+1:

Comment (drvinceknight) [inline axelrod/moran.py]: Any reason for the :-1: reaction @souravsingh ?

**Comment by marcharper** [comment]
@janga1997 Thanks for helping with the review! We try to keep things positive and it's not necessary to thumbs down lines. Just leave a comment as to why you think the line should be changed, and an alternate version if possible

**Comment by janga1997** [comment]
@marcharper I didn't realise I added a thumbs down reaction to any line. I just pointed out where to make changes.

**Comment by drvinceknight** [comment]
> @marcharper I didn't realise I added a thumbs down reaction to any line. I just pointed out where to make changes.

Indeed you didn't add any reaction @janga1997.

@marcharper if you hover over a reaction you can see who posted it. In this case the thumbs down reactions were posted by @souravsingh which is why I asked:

> Any reason for the :-1: reaction @souravsingh ?

File: axelrod/moran.py (hunk 1 above)
Comment (souravsingh) [inline axelrod/moran.py]: @drvinceknight A full review for the PR wasn't done. I had missed many other types in the PR, hence the Thumbs down.

Comment (drvinceknight) [inline axelrod/moran.py]: @souravsingh as @marcharper said below we try to keep things positive here.

**Comment by marcharper** [comment]
Oh, my mistake. Carry on!

**Comment by souravsingh** [comment]
@marcharper I believe the PR can be merged(unless there is something I am missing)

File: axelrod/moran.py (hunk 1 above)
Comment (drvinceknight) [inline axelrod/moran.py]: @souravsingh can you add the requested return type for this function (and the others)?

**Comment by souravsingh** [comment]
Is the PR ready for merging?

**Comment by drvinceknight** [comment]
> Is the PR ready for merging?

It looks right to me. Thanks :)

**Comment by drvinceknight** [comment]
Just realised that there are a number of type tests that are failing on this, I'm not sure how we missed it. I'm reverting the PR.

**Comment by drvinceknight** [comment]
Have opened #868

**Pull request merged**